rest/src/main/java/fi/otavanopisto/pyramus/rest/ApplicationRESTService.java

```diff
@@ -0,0 +1,472 @@
+import fi.otavanopisto.pyramus.domainmodel.application.ApplicationState;
+import fi.otavanopisto.pyramus.domainmodel.base.Language;
+import fi.otavanopisto.pyramus.domainmodel.base.Municipality;
+import fi.otavanopisto.pyramus.domainmodel.base.Nationality;
+import fi.otavanopisto.pyramus.domainmodel.base.School;
+import fi.otavanopisto.pyramus.domainmodel.system.Setting;
+import fi.otavanopisto.pyramus.domainmodel.system.SettingKey;
+import fi.otavanopisto.pyramus.rest.annotation.Unsecure;
+import net.sf.json.JSONException;
+import net.sf.json.JSONObject;
+
+@Path("/application")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+@Stateful
+@RequestScoped
+public class ApplicationRESTService extends AbstractRESTService {
+
+  private static final Logger logger = Logger.getLogger(ApplicationRESTService.class.getName());
+
+  @Context
+  private UriInfo uri;
+
+  @Inject
+  private SchoolDAO schoolDAO;
+
+  @Inject
+  private MunicipalityDAO municipalityDAO;
+
+  @Inject
+  private NationalityDAO nationalityDAO;
+
+  @Inject
+  private LanguageDAO languageDAO;
+
+  @Path("/createattachment")
+  @POST
+  @Unsecure
+  @Consumes(MediaType.MULTIPART_FORM_DATA + ";charset=UTF-8")
+  public Response createAttachment(MultipartFormDataInput multipart, @HeaderParam("Referer") String referer) {
+    if (!isApplicationCall(referer)) {
+      return Response.status(Status.FORBIDDEN).build();
+    }
+    String attachmentsFolder = getSettingValue("application.storagePath");
+    try {
+      byte[] fileData = getFile(multipart, "file");
+      String name = getString(multipart, "name");
+      String attachmentFolder = getString(multipart, "applicationId");
+      if (fileData == null || fileData.length == 0 || name == null || attachmentFolder == null) {
+        logger.log(Level.SEVERE, "Application attachment preconditions not met");
+        return Response.status(Status.INTERNAL_SERVER_ERROR).build();
+      }
+      File folder = Paths.get(attachmentsFolder, attachmentFolder).toFile();
+      if (!folder.exists()) {
+        folder.mkdir();
+      }
+      File file = Paths.get(attachmentsFolder, attachmentFolder, name).toFile();
+      if (file.exists()) {
+        return Response.status(Status.CONFLICT).build();
+      }
+      FileUtils.writeByteArrayToFile(file, fileData);
+      logger.log(Level.INFO, String.format("Stored application attachment %s", file.getAbsolutePath()));
+    }
+    catch (Exception e) {
+      logger.log(Level.SEVERE, "Failed to store application attachment", e);
+      return Response.status(Status.INTERNAL_SERVER_ERROR).build();
+    }
+    return Response.noContent().build();
+  }
+  
+  @Path("/removeattachment/{ID}")
+  @DELETE
+  @Unsecure
+  public Response removeAttachment(@PathParam("ID") String applicationId, @QueryParam("attachment") String attachment, @HeaderParam("Referer") String referer) {
+    try {
+      if (!isApplicationCall(referer)) {
```

Comment:
Referer might be unreliable but not really sure if worth changing. 

https://en.wikipedia.org/wiki/HTTP_referer -> "some web browsers allow the user to disable the sending of referrer information" also it could be faked for malicious purposes.